File: mobile/android/sdk/src/main/kotlin/com/yuledkcs/sdk/hub/SDKConfig.kt
package com.yuledkcs.sdk.hub

/**
 * yuleDKCS SDK 配置
 */
data class SDKConfig(
    /// Hub REST Gateway 主机地址（如 "hub.yuletech.com"）
    val hubEndpoint: String = "hub.yuletech.com",
    /// Hub REST Gateway 端口（默认 8080）
    val hubPort: Int = 8080,
    val enableLogging: Boolean = false,
    val platform: Platform = Platform.ANDROID
)

enum class Platform {
    UNSPECIFIED, IOS, ANDROID, HARMONY
}

File: mobile/ios/Sources/YDKHubClient/SDKConfig.swift
import Foundation

/// yuleDKCS SDK 配置
public struct SDKConfig {
    /// Hub REST Gateway 主机地址（如 "hub.yuletech.com"）
    let hubEndpoint: String
    /// Hub REST Gateway 端口（默认 8080）
    let hubPort: Int
    /// 手机平台（SDK 自动检测）
    let platform: Platform
    /// 是否启用日志
    let enableLogging: Bool

    public init(
        hubEndpoint: String,
        hubPort: Int = 8080,
        platform: Platform = .iOS,
        enableLogging: Bool = false
    ) {
        self.hubEndpoint = hubEndpoint
        self.hubPort = hubPort
        self.platform = platform
        self.enableLogging = enableLogging
    }
}

/// 手机平台
public enum Platform: Int, Codable {
    case unspecified = 0
    case iOS = 1
    case android = 2
    case harmony = 3
}
